promql/functions.go

```diff
@@ -872,18 +872,26 @@ func funcLabelJoin(ev *evaluator, args Expressions) model.Value {
 
 	outSet := make(map[model.Fingerprint]struct{}, len(vector))
 	for _, el := range vector {
-		srcVals := make([]string, 0)
-		for _, src := range srcLabels {
-			srcVals = append(srcVals, string(el.Metric.Metric[src]))
+		srcVals := make([]string, len(srcLabels))
+		issrcValid := false
+		for i, src := range srcLabels {
+			strval := string(el.Metric.Metric[src])
+			srcVals[i] = strval
+			if strval != "" {
+				issrcValid = true
```

Comment:
This logic doesn't seem right, even with all empty labels a non-empty seperator can produce non-empty output.

I'd do the join and then check if that's non-empty.